### ISSUE TITLE
details for RTC additions  from coga

### ISSUE THREAD

**Issue by lseeman** [w3c/apa#172]
Thank you for offering us this time to make clear specific requests to RTC Accessibility User Requirements.
 We have tried to complete it as soon as we could!
Please let us know if you have questions or would like further discussion. 

**Add to User Need 5:**  Also, a person using an AAC device may need to use their device and their voice/microphone simultaneously to manage two types of audio (one from their voice, one from their device). Note: this may or may not be a person with a cognitive disability, who needs the other supports listed for those with cognitive disabilities.

**Add to User Need 11:** Also, a person using an AAC device may also require the ability to send information from their device during an emergency.
 
**Add to User Need 16** the following requirements:
-REQ 16c: Allow support from any point in the process.
-REQ 16d: The support is easy to find and use such that people with learning and cognitive disabilities can  give feedback, ask questions, and get feedback:
-- in a similar time frame to everyone else,
-- using their preferred communication method (form, email, chat, phone support, etc.)
-- know how to get help or information 
-REQ 16e: When human help is available people with learning and cognitive disabilities know how to get human help and can manage the process easily. This includes allowing a user to  find a human by pressing a common reserved digit or term (typically the word help, or the number 0).
-REQ 16f: Allow additional third party support such as speech-speech-relay-service 

**We would also like to add a user need (with requirements) to make the system useable by people with  learning and cognitive disabilities  **

**User Need xx:** Users with memory impairments, learning and cognitive disabilities need usable systems that they can manage independently.

- REQ xx.a: Pauses are  between phrases in order to allow processing time of language and options. The system allows for slow speakers, stutters, repetitions, and corrections of terms by the user.
- REQ xx.b: Options in text should be given before the term or digit to select, or the instruction to select that option. This will mean that the user does not need to remember the digit or instruction whilst processing the term. For example: The prompt “press 1 for the secretary,” requires the user to remember the digit 1 while interpreting the term “secretary”. A better prompt is “for the secretary (pause): press 1” or “ for the secretary (pause) or for more help (pause): press 1”.
- REQ xx.c: Error recovery is simple, without users having to start at the beginning . Users can consistently go back and undo previous steps if they make a mistake. The system should offer the user more support and/or a human operator if the error persists. Error responses should not end the call or send the user to a more complex menu.
- REQ xx.d: Advertisements and other extraneous information should not be read as it can confuse the user and can make it harder to retain attention. Distractions and background noise are avoided.
- REQ xx.f: Help the user know where they are and restore context. This includes if the user loses focus and needs to restore context in the middle of the process.
- REQ xx.g: Help  the user understand the architecture of the process, so that they understand where they are and can feel oriented in the process. Streamline processes and workflows so that they include only the minimal necessary steps. Separate out optional steps that are supplemental but not required. Do not require the user to go through optional steps.
- REQ xx.h: Language and terms used are clear and as simple and jargon-free as possible.Use easy to understand language, and make sense to people with limited vocabulary and language impairments. If responses are limited, help the user  identify the right words and the terms should be common terms that  people with learning and cognitive disabilities would be likely to use.
- REQ xx.i: Use Tapered Prompts. Best practices in voice user interface design include providing several different prompts for each point in the interaction. The different prompts are used based on the user’s behavior. For example, if the user takes a long time to respond to a prompt, a simpler or more explanatory version of the prompt can be used instead of the default. They should be used to increase the level of prompt detail when the user does not respond as expected.
- REQ xx.j: Use usability best practices for voice menus. Simple-to-navigate voice-menu systems with limited options that make sense to people with limited vocabulary and language impairments, so that users do not struggle with multiple steps and can identify options quickly. 
- REQ xx.k: Let the user know at the start of the process, everything that they may need to complete the process, such as social security number, or other identification. Do not assume users can recall this information or find it easily. Notify the user about all charges at the start of a transaction including typical values. 
- REQ xx.l: Avoid memory barriers such as: 
- navigating voice menus that involve remembering a specific number or term,
- remembering numbers while processing words on a voice menu,
- transcribing text, or
- remembering passwords.
- REQ xx.m: Avoid timeouts and let the user save their work as they go. When this is not possible, inform the user when they initiate the process the amount of time available to complete the process, and  if the user will lose entered data if a timeout occurs.

sources include: [Making Content Usable for People with Cognitive and Learning Disabilities](https://www.w3.org/TR/coga-usable/), [Cognitive Accessibility Roadmap and Gap Analysis](https://rawgit.com/w3c/coga/master/gap-analysis/), [voice interfaces issue paper](https://w3c.github.io/coga
/issue-papers/voice-menus.html)
, [conversation interfaces issue paper -draft](https://docs.google.com/document/d/1B1vCqlU1IF5UmqxhJAy8Khdi-kRQNPalVX8f3lCMr7w/edit#heading=h.9j8rnhzb9sv8)

Thank you so much for your consideration.

The COGA TF

**Comment by RealJoshue108** [comment]
Thanks for this @lseeman - it's welcome and very useful. We will be discussing in RQTF.

**Comment by RealJoshue108** [comment]
@lseeman RQTF walked thru these suggestions. A large part of these are not relevant to the RAUR, but would be to the Natural Language Interface A11y document that we are working on, so we have noted those and relabelled an issue with NAUR. Many of the rest we feel are either general requirements, or essentially WCAG/Silver issues, so should be discussed there.  

Much of what we are trying to relate to WebRTC group, is about browser support and not application support. Some specific parts that were relevant and we have added to the RAUR relate to the reference to AAC in User Needs 5 & 11, as well as the speech-speech-relay-service, which we have updated a relevant requirement with.

We do appreciate this useful feedback.
